tools/auth/authentik.go

```diff
@@ -0,0 +1,34 @@
+package auth
+
+/*
+ *	https://goauthentik.io/
+ *  authentik is an open-source Identity Provider focused on flexibility and versatility
+ */
+
+import (
+	"fmt"
+
+	"golang.org/x/oauth2"
+)
+
+var _ Provider = (*Authentik)(nil)
+
+// Name of the oauth provider
+const NameAuthentik string = "authentik"
+
+type Authentik struct {
+	*baseProvider
+}
+
+func NewAuthentikProvider() *Authentik {
+	return &Authentik{&baseProvider{
+		scopes:   []string{"profile"},
```

Comment:
Correct me if I am wrong please, but I feel that scopes can be preconfigured on provider side. If that's the case, it might make sense to not request specific scopes, as it will override the provider's configuration (while pocketbase request scopes are not configurable atm).

Ref: https://goauthentik.io/docs/providers/oauth2/#scope-authorization

Reply:
You are partially correct. This default implementation just requests a scope that is enabled by default in Authentik, but ofc this could be configured in any way on the provider side.
I am struggling at the moment to find where the frontend settings are passed into this class. If someone could give me a hint that would be very helpful for the progress of this PR.

Reply:
> I am struggling at the moment to find where the frontend settings are passed into this class. If someone could give me a hint that would be very helpful for the progress of this PR.

I'm not sure if that's what you are asking, but users can't configure the default `scopes` from the Admin UI at the moment. At the end, they are just minimal defaults for convenience and are used only in the login url(s) that are returned with the [`List auth method` endpoint](https://pocketbase.io/docs/api-records/). Users can adjust/replace them client-side by modifing the `scope` query parameter in the provider login url(s).

Reply:
Thanks @ganigeorgiev 

- What I meant is that Authenktik.go uses a *baseProvider instance, but I dont see where in the baseProvider instance I can retrieve the settings that get set in the Svelte frontend component AuthentikOptions.svelte. This is essential, since all the endpoints get configured there (and authentik is self-hosted, so I can not provide a general URL in the constructor)
- Maybe you could also give me a hint since authentik is a very general provider, and could also allow for example Keycloak to be used, if we want to keep it named Authentik or if we want to give it a more general name.
- I need some feedback about the frontend part

Reply:
@pr0ton11 The Admin UI settings API doesn't operate directly with the `auth.baseProvider` and instead uses the `settings.AuthProviderConfig` - https://github.com/pocketbase/pocketbase/blob/c1921aeef8b28f9df66d5be3143e0c39dd079bec/models/settings/settings.go#L461-L468

You can find details about the available settings API fields in https://pocketbase.io/docs/api-settings/.

Reply:
Thanks for helping with the PR, I completed the implementation and it looks like everything is working as expected now